File: 240131/격자 위의 편안한 상태/comfortable-state-on-the-grid.py
n,m=map(int,input().split())

dxs,dys=[0,1,0,-1],[1,0,-1,0]

grid=[[0] * n for _ in range(n)]

def in_range(x,y):
    return 0<=x<n and 0<=y<n

def check(x,y):
    cnt=0
    for dx,dy in zip(dxs,dys):
        nx,ny=x+dx,y+dy
        if in_range(nx,ny):
            if grid[x+dx][y+dy]!=0:
                cnt+=1
    return cnt==3

for i in range(m):
    x,y=map(int,input().split())
    x-=1
    y-=1
    grid[x][y]=1
    if check(x,y):
        print(1)
    else:
        print(0)

File: 260103/격자 위의 편안한 상태/comfortable-state-on-the-grid.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const moves = input.slice(1).map((move)=>move.split(' ').map(Number));

const grid = Array.from({length:n+1}, ()=>Array(n+1).fill(0));

const dx=[0,1,0,-1], dy = [1,0,-1,0];

function isGrid(x,y){
    return 1<=x && x<=n && 1<=y && y<=n;
}

for(let move of moves){
    const [x,y] = move;
    grid[x][y] = 1;
    let cnt = 0;
    for(let i=0; i<4; i++){
        const nx = x+dx[i], ny = y+dy[i];
        if(isGrid(nx,ny) && grid[nx][ny]===1){
            cnt++;
        }
    }
    console.log(cnt===3?1:0);
}
